src/main/java/me/jellysquid/mods/lithium/common/entity/EntityClassGroup.java

```diff
@@ -0,0 +1,119 @@
+package me.jellysquid.mods.lithium.common.entity;
+
+import net.minecraft.entity.Entity;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Class for grouping Entity classes that meet some requirement for use in TypeFilterableList
+ * Designed to allow create groups of entity classes that are updated when mods add new entities that fit into the group.
+ * @author 2No2Name
+ */
+public class EntityClassGroup {
+    //Keep a set of classes that were already added to matching class groups, so we only analyse them once.
+    private static final ConcurrentHashMap<Class<?>, Object> knownEntityClasses = new ConcurrentHashMap<>(); //value unused, no set variant available
+    //Keep track of available class groups for updating them in case an entity class is instantiated for the first time
+    private static final List<EntityClassGroup> entityClassGroups = new ArrayList<>();
+
+    public static final EntityClassGroup COLLISION_BOX_OVERRIDE = new EntityClassGroup(
+            (Class<?> entityClass) -> {
+                boolean overwritten;
+                while(entityClass != null && entityClass != Entity.class) {
+                    try {
+                        overwritten = true;
+                        entityClass.getDeclaredMethod("method_30948");
+                    } catch (NoSuchMethodException e) {
+                        overwritten = false;
+                        entityClass = entityClass.getSuperclass();
+                    }
+                    if (overwritten) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+    );
+    public static final EntityClassGroup HARD_COLLISION_BOX_OVERRIDE = new EntityClassGroup(
+        (Class<?> entityClass) -> {
+            boolean overwritten;
+            while(entityClass != null && entityClass != Entity.class) {
+                try {
+                    overwritten = true;
+                    entityClass.getDeclaredMethod("method_30949", Entity.class);
```

Comment:
```suggestion
                    entityClass.getDeclaredMethod(FabricLoader.getInstance().getMappingResolver().mapMethodName("intermediary", "net/minecraft/class_1297", "method_30949", "(Lnet/minecraft/class_1297;)Z", Entity.class);
```

Reply:
Added that now. Thanks for the suggestion